vm/devices/get/guest_emulation_transport/src/client.rs

```diff
@@ -90,85 +88,55 @@ impl GuestEmulationTransportClient {
         self.version
     }
 
-    /// Sends a VMGS read request over the GET device
+    /// Reads `sector_count` sectors of size `sector_size` from the VMGS disk,
+    /// starting at `sector_offset`.
     ///
-    /// # Arguments
-    /// * `sector_offset` - Offset to start reading from the file
-    /// * `buf` - Buffer to read the VMGS data into
-    /// * `sector_size` - Size of a sector
+    /// The caller must ensure the read is smaller than the maximum transfer
+    /// size.
     pub async fn vmgs_read(
```

Comment:
I like this change a lot. The fact the GET client had this sort of non-trivial chunking logic was an unfortunate wart, so bringing these APIs more in-line with "the GET client is a thin layer to interact with the GET" is good to see.